packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts

```diff
@@ -4,20 +4,57 @@ import { fetchToCurl } from "../helper";
 import { Context } from "../index";
 import { Rule } from "../rule";
 
+function compare(
+  expected: { [k: string]: boolean | string | number },
+  actual: typeof expected
+) {
+  return Object.keys(expected).reduce(
+    (acc, k) => {
+      if (!actual[k]) {
+        acc[k] = `expected: [${expected[k]}], actual: property missing`;
+        return acc;
+      } else if (expected[k] !== actual[k]) {
+        acc[k] = `expected: [${expected[k]}], actual: [${actual[k]}]`;
+        return acc;
+      } else {
+        return acc;
+      }
+    },
+    {} as typeof expected
+  );
+}
+
+async function urlHasContentType(
+  url: string,
+  headers: Context["headers"],
+  contentType: string
+) {
+  const res = await fetch(url, { headers });
+  if (!res.ok) {
+    return false;
+  }
+  return res.headers.get("content-type") === contentType;
+}
+
+const REQUEST_HEADERS = {
+  "accept": "application/signed-exchange;v=b3",
+  "amp-cache-transform": `google;v="1"`
+};
+const EXPECTED_VERSION = "1b3";
+
 export class SxgDumpSignedExchangeVerify extends Rule {
   async run({ url, headers }: Context) {
```

Comment:
suggestion: split the method up into multiple methods, where each method implements a different check. Currently, it's very hard to determine from the code what is checked.

Reply:
I had a few goes and couldn't figure out a way to improve it (with the current reporting structure). Have added in some whitespace to create sections and will think a bit more about it. I'm not happy with this either.